frame/vesting/src/lib.rs

```diff
@@ -554,13 +561,22 @@ impl<T: Config> Pallet<T> {
 	}
 
 	/// Returns balance available to claim for a given account, asset and vesting schedules, based
-	/// on current block number
+	/// on current block number, as well as the claimed balance per vesting schedule.
 	/// It also updates the already claimed balance and removes completely vested schedules
 	fn unlocked_claimable_balance(
 		who: &AccountIdOf<T>,
 		asset: AssetIdOf<T>,
 		vesting_schedule_ids: VestingScheduleIdSet<T::VestingScheduleId, T::MaxVestingSchedules>,
-	) -> Result<BalanceOf<T>, DispatchError> {
+	) -> Result<
+		(BalanceOf<T>, BoundedBTreeMap<T::VestingScheduleId, BalanceOf<T>, T::MaxVestingSchedules>),
+		DispatchError,
+	> {
+		let mut amount_claimed: BoundedBTreeMap<
+			T::VestingScheduleId,
+			BalanceOf<T>,
+			T::MaxVestingSchedules,
+		> = BoundedBTreeMap::new();
```

Comment:
NIT: perhaps out of the scope of this PR, but this function is effectively a `reduce` to the `amount_claimed` and `total_balance_to_claim`, and could be refactored to be much more clear and readable in that respect (in my opinion).

Reply:
I agree that this can be rewritten as a `reduce` but for the sake of having these events available for Subsquid, I'd rather get this merged and then refactor it on another PR, if that's not a problem. I can create this other PR tomorrow, as it shouldn't take long.

Reply:
Yeah absolutely, I just wanted to make note of it 👍